### PR TITLE
Extend support for minishift profile feature

### PR DESCRIPTION
**Pull request by odockal**
- Testing of server adapter operatinos only with CDK 3.2 version
- UI testing extended with Server Wizard test for CDK 3.2+
- UI testing extended with Server Editor test for CDK 3.2+
- Deprecated CDKIntegrationTest
- Deprecated CDK3IntegrationTest
- Improvement of waiting in new server wizard

Signed-off-by: Ondrej Dockal <odockal@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: test-framework/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/server/ui/wizard/NewCDKServerWizard.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
+
+/**
+ * Class represents New Server Wizard, it is required because
+ * here it can be overriden finish and cancel that only kills Refreshing Server adapter list job 
+ * when such job has started unlike is implemented in NewServerWizard class
+ * @author odockal
+ *
+ */
+public class NewCDKServerWizard extends NewMenuWizard {
```
[the remark this answers is not in the document]
Comment (odockal) [reply test-framework/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/server/ui/wizard/NewCDKServerWizard.java]: @adietish I know that this looks as a bad design, but NewCDKServerWizard is just extending NewMenuWizard which comes from reddeer, I cannot do much about it.

[the remark this answers is not in the document]
Comment (adietish) [reply test-framework/org.jboss.tools.cdk.reddeer/src/org/jboss/tools/cdk/reddeer/server/ui/wizard/NewCDKServerWizard.java]: I know, so I made "thumbs down" to sonar :)

File: itests/org.jboss.tools.cdk.ui.bot.test/README.md [hunk 2]
```diff
@@ -6,48 +6,65 @@ To be able to run CDK integration tests, couple of pre-requisities must be fulfi
...
 * `-Pcdk3-all-tests` (CDK3AllTestsSuite)
+* `-Pcdk32-all-tests` (CDK32AllTestsSuite)
 
 To run tests, type this command:
 ```
 mvn clean verify -Dtest.installBase=/path/to/ide/to/run/against
 -Dminishift.hypervisor=kvm|virtualbox|hyperv|xhyve
--Dminishift.path=/path/to/your/minishift/binary
+-Dminishift=/path/to/your/cdk-3.1/minishift
+-Dminishift.profile=/path/to/your/cdk-3.2/minishift
```
Comment (adietish) [inline itests/org.jboss.tools.cdk.ui.bot.test/README.md]: is it possible to run cdk31 and cdk32 tests at the same time?

**Comment by adietish** [comment]
I have failing tests, they're not consistent though:
```
Failed tests:
  CDK32ServerEditorTest.testCDK32ServerEditor:84->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3.2 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK32ServerEditorTest.testCDK32Hypervisor:102->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3.2 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK32ServerEditorTest.testInvalidMinishiftLocation:111->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3.2 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK3ServerEditorTest.testCDK3ServerEditor:80->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK3ServerEditorTest.testInvalidMinishiftLocation:107->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK3ServerEditorTest.testCDK3Hypervisor:98->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK3ServerEditorTest.testCDK3ServerEditor:80->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK3ServerEditorTest.testInvalidMinishiftLocation:107->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK3ServerEditorTest.testCDK3Hypervisor:98->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK32ServerEditorTest.testCDK32ServerEditor:84->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3.2 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK32ServerEditorTest.testCDK32Hypervisor:102->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3.2 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled
  CDK32ServerEditorTest.testInvalidMinishiftLocation:111->CDKServerEditorAbstractTest.assertCDKServerWizardFinished:106 Fails to create Container Development Environment 3.2 Server via New Server Wizard due to Exception occured in CDK server wizard, wizard was canceled

Tests in error:
  Given key Minishift path value is null
  Given key Minishift path value is null
  Given key Minishift path value is null
  Given key Minishift path value is null
  Given key Minishift path value is null
  Given key Minishift path value is null
  Given key Minishift path value is null
  Given key Minishift path value is null
  Given key Vagrantfile path value is null
  Given key Minishift path value is null
  Given key Minishift path value is null
  Given key Vagrantfile path value is null
  Given key Minishift path value is null
  Given key Minishift path value is null
  Given key Minishift path value is null

Tests run: 27, Failures: 12, Errors: 15, Skipped: 0

[ERROR] There are test failures.
```

**Comment by adietish** [comment]
we should also state in the README so that it explicitly says to use -PITtests. And also say that both 31 and 32 are run, but in 31 only UI is tested, the minishift is not started/stopped since 32 and 31 are mutually exclusive

**Comment by adietish** [comment]
```
mvn clean verify -Dminishift.hypervisor=xhyve -Dminishift=/Applications/cdk3/cdk-311 -Dminishift.profile=/Applications/cdk3/cdk-320 -Ddevelopers.username=adietish@redhat.com -Ddevelopers.password=1q5t9o3e -Dusage_reporting_enabled=false -PITest -Pcdk32-all-tests
```
doesn't run the tests, it just compiles the code, wont run them, so apparently it does revert the ```-DskipITests``` to ```true```

**Comment by jeffmaury** [comment]
Should be -PITests
Hope the password is fake

**Comment by jeffmaury** [comment]
I run both CDK32AllTestsSuite or CDK3AllTestsSuite successfully. However, I have the feeling that when you run the CDK3AllTestsSuite you must reset the .minishift folder because if you don't then start will fail trying to download oc. So this should be specified in the readme

**Comment by odockal** [comment]
@jeffmaury Yes, that is true. I will put this info into readme as well.
@jeffmaury @adietish I have an idea, I could possibly "fake" vagrantfile and minishift (cdk 3.1.1) binary and place them into tests themselves (into resources folder for example), to remove the need of downloading those two component, when running CDKAllTestSuite, CDKSmokeTestsSuite and CDK32AllTestsSuite. One will still need to setup CDK-3.1.1 if he want to run CDK3AllTestSuite... wdyt?

**Comment by adietish** [comment]
running with ```-PITests -Pcdk32-all-tests``` had me running CDK3AllTestsSuite and CDKAllTestSuite obviously running vagrant and cdk31 tests. If I am a dev that only has cdk32 installed, how do I do to have only the ITs for cdk32 running?

**Comment by adietish** [comment]
@odockal @jeffmaury I dont think that requiring to reset .minishift before running a suite is a good idea and should be avoided if possible. If this requirement currently exists I'd highly advocate the faking that you suggested.

**Comment by odockal** [comment]
@adietish @jeffmaury faking the CKD-3.1 binary and Vagrant file could be used for CDKAlltestSuite/Smoke and CDK32 suites.

Only changes would be that test suite will not require passing path to vagrantfile and cdk-3.1 and it will take, let's say, Vagrantfile from latest CDK 2.x and minishift (v1.3.1 - should be working the same way as cdk-3.1.1) binary which is only 3 MB big, and those files would be placed in itests resource folder...

And to make it more clear, "fake" files would be used to testing UI only. Not for real testing of CDK 3.2+ = start and stop it via server adapter. 

@adietish I really do not know, how to force user, to have cdk in proper state other than in form of 'note'. This CDK (3.2+ now) is supposed to be used for testing of the adapter = verification that server adapter from CDK tooling in IDE can actually operate with CDK via its server adapter. This CDK cannot be faked.

**Comment by odockal** [comment]
@adietish base on what do you judge that not only CDK32AllTestSuite test were run but also CDKAllTestsSuite? I have just tried to run -Pcdk32-all-tests and it has run only CDK32 test classes. I will just add implicit Vagrantfile and minishift, but it should have run only these CDK32 test classes.

**Comment by jeffmaury** [comment]
Are you sure we can fake CDK3.1 binary for CDK3.2 itests. I think the server adapter wizard now checks that the minishift binary has the proper version as we now have 2 server adapters for CDK3

**Comment by odockal** [comment]
I think I can use upstream minishift binary (v1.3.1) which has only 3 MB or so. Otherwse, I can write short runnable script/exe returning proper output. I am testing it now.

**Comment by odockal** [comment]
Well.. I can only make binaries/runnable files that would be writing out proper output, I cannot use upstream minishift binaries as validation focus on string CDK v3.x.x from 'minishift version' and that is something what minishift does not have...

**Comment by adietish** [comment]
@odockal after running i had surefire failure reports for CDK3 and CDK suites.

**Comment by odockal** [comment]
I just have checked, that this is required amount of arguments, needed for running CDK32AllTestsSuite:
```
mvn clean verify -Dusage_reporting_enabled=false -Ddevelopers.username=username -Ddevelopers.password=password -Dminishift.profile=/home/odockal/Programs/minishift/cdk-3.2.0/minishift -Dminishift.hypervisor=kvm -Dminishift=/home/odockal/Programs/minishift/cdk-3.1.1/minishift -PITests -Pcdk32-all-tests 
```

Unfortunately, it is very uncomfortable for other users that would like to run CDK32 only tests, that they need to have minishift argument set ( path to cdk-3.1.1 binary). This is because UI of CDK 3.x is tested with both variants of CDK (3.1.1 vs 3.2). It could be solved in future, as I suggested above. WDYT @adietish ?

**Comment by adietish** [comment]
finally after correcting the launch params to 
```
mvn clean verify -Dminishift.hypervisor=xhyve -Dminishift=/Applications/cdk3/cdk-311 -Dminishift.profile=/Applications/cdk3/cdk-320 -Ddevelopers.username=adietish@redhat.com -Ddevelopers.password=XXXX -Dusage_reporting_enabled=false -PITests -Pcdk32-all-tests
```
and recompiling all of openshift (since nexus doesnt deliver), I had it running with a single failure:
```
Failed tests:
  CDK32ServerAdapterConnectionTest.testCDK3ServerAdapterConnection:44->CDKServerAdapterAbstractTest.testOpenshiftConncetion:271 Could not find deployed sample OpenShift project
```
image

**Comment by jeffmaury** [comment]
Got this error but fixed it because you must keep the default 'My Project' project and not delete it. Maybe worth noting it in the readme file

**Comment by adietish** [comment]
@jeffmaury I tend to think that we should "fix" this. It's far too likely that you have things changed in your server and far too unexpected if it then fails because of this.

**Comment by odockal** [comment]
@adietish @jeffmaury You mean to fix this in itests or in tooling? I am not sure how can I check in tests something what user has changed himself? How can this be replicated, so that OS connection has some other project than My Project?

**Comment by adietish** [comment]
@odockal yes I believe that we should fix in the test. let me ask differently: what value is there in the failure where minishift works, connection works too, but the project (if there's any) is not "My Project" but "node" (or anything else)? 
I think that we can simply drop the assertion for the project. Refresh on the connection would fail if minishift wasnt up.

**Comment by odockal** [comment]
@adietish I agree, If it is enough to refresh connection to check whether if it works or not I would remove the assertion, as you suggested.

**Comment by adietish** [comment]
@odockal great, thx. testing

**Comment by adietish** [comment]
+1 from me:
```
Results :

Tests run: 8, Failures: 0, Errors: 0, Skipped: 0
```

**Comment by adietish** [comment]
we're currently frozen so we can merge this as soon as master is open again.

**Comment by odockal** [comment]
Yeah, that is right. Thanks!

**Comment by odockal** [comment]
@adietish master is unfrozen now, can we merge the PR? Thanks!

**Comment by odockal** [comment]
@jeffmaury @adietish Do you mind to merge the PR since it got +1? Thanks!

**Comment by odockal** [comment]
Oh, I rebased the branch and it got broken by some sonar error...

**Comment by adietish** [comment]
jenkins is failing because of unknown reasons. Running the tests locally shows 0 errors:

```
Results :

Tests run: 8, Failures: 0, Errors: 0, Skipped: 0
```
merging. Thanks!

**Pull request merged**